middlewares/validation-schama.ts

```diff
@@ -0,0 +1,114 @@
+    .typeError('Dozwolone są tylko liczny naturalne w polu ilość miesięcy doświadczenia komercyjnego')
+    .notRequired(),
+});
+
+export const registerHrSchema = yup.object().shape({
+  email: yup.string().email('Proszę podać poprawny adres email.').required('Email jest wymagany do rejestracji.'),
+  fullName: yup.string().required('Imię i nazwisko jest wymagane do rejestracji.'),
+  company: yup.string().required('Nazwę firmy jest wymagane do rejestracji.'),
+  maxReservedStudents: yup
+    .number()
+    .min(1, 'Minimalna liczba osób to 1.')
+    .max(999, 'Maksymalna liczba osób 999.')
+    .required('Podaj maksymalną liczbę osób umówionych na rozmowę.'),
+});
+
+export const setStatusSchema = yup.object().shape({
+  email: yup.string().email('Proszę podać poprawny adres email.').required('Email jest wymagany zmiany statusu.'),
+  status: yup.string().oneOf(Object.values(Status), 'Nie prawidłowy status.').required('Przekaż status do zmiany.'),
+});
+
+export const registerUserSchema = yup.object().shape({
+  email: yup.string().email('Proszę podać poprawny adres email.').required('Email jest wymagany do rejestracji.'),
+  phone: yup.string().notRequired(),
+  firstName: yup.string().required('Imię jest wymagane do rejestracji.'),
+  lastName: yup.string().required('Nazwisko jest wymagane do rejestracji.'),
+  githubUsername: yup.string().required('Nazwa użytkowania github jest wymagane do rejestracji.'),
+  portfolioUrls: yup.array().of(yup.string().url('Nie poprawny adres url do Twojego portfolio')).notRequired(),
+  projectUrls: yup
+    .array()
+    .of(yup.string().matches(/^https?:\/\/(www\.)?github\.com\/.*$/, 'Nie poprawny adres url projektu końcowego'))
+    .required()
+    .min(2, 'Wymagane jest podanie adresów do projektu na FE i BE')
+    .required('Dodaj adresy url do projektu końcowego'),
+  bio: yup.string().notRequired(),
+  expectedTypeWork: yup
+    .string()
+    .oneOf(Object.values(TypeWork), 'Nie prawidłowe dane w polu oczekiwane miejsce pracy.')
+    .required('Wybierz preferowane miejsc pracy'),
+  targetWorkCity: yup.string().notRequired(),
+  expectedContractType: yup
+    .string()
+    .oneOf(Object.values(ContractType), 'Nie prawidłowe dane w polu typ kontraktu')
+    .required('Wybierz preferowane typ kontraktu'),
+  expectedSalary: yup.string().notRequired(),
+  canTakeApprenticeship: yup
+    .string()
+    .oneOf(['TAK', 'NIE'], 'Wybierz prawidłowe dane: "TAK" lub "NIE"')
+    .required('Wybierz czy zgadzasz się na odbycie bezpłatnego stażu.'),
+  monthsOfCommercialExp: yup
+    .number()
+    .min(0, 'Minimalna liczba miesięcy doświadczenia komercyjnego to 0')
+    .integer('Tylko liczby naturalne w polu ilość miesięcy doświadczenia komercyjnego.')
+    .typeError('Dozwolone są tylko liczny naturalne w polu ilość miesięcy doświadczenia komercyjnego')
+    .notRequired(),
+  education: yup.string().notRequired(),
+  workExperience: yup.string().notRequired(),
+  courses: yup.string().notRequired(),
+});
```

Comment:
Super! Brawo, że to jest osobno. Można nawet zrobić per kontroler osobny plik. I btw, czemu to jest w folderze `middlewares`? Może lepiej `validators` albo coś takiego?